front/src/modules/trainschedule/components/Timetable/PacedTrain/PacedTrainItem.tsx

```diff
@@ -74,12 +73,18 @@ const PacedTrainItem = ({
   const toggleOccurrencesList = () => setIsOccurrencesListOpen((open) => !open);
   const selectPathProjection = async () => {};
 
-  const deletePacedTrain = async () => {
-    // if (isSelected) {
-    //   // we need to set selectedTrainId to undefined, otherwise just after the delete,
-    //   // some unvalid rtk calls are dispatched (see rollingstock request in SimulationResults)
-    //   dispatch(updateSelectedTrainId(undefined));
-    // }
+  const selectOccurrence = useCallback((occurrenceId: TrainId) => {
+    dispatch(updateSelectedTrainId(occurrenceId));
+  }, []);
+
+  const hasOccurrenceSelected = selectedTrainId?.includes(pacedTrain.id);
```

Comment:
`hasOccurrenceSelected` is only used in deletePacedTrain, so it can be instantiated inside the function ;)